crates/rome_js_analyze/src/analyzers/nursery/no_misused_new.rs

```diff
@@ -0,0 +1,232 @@
+    TsReferenceType, TsTypeAliasDeclaration,
+};
+use rome_rowan::{declare_node_union, AstNode, SyntaxToken, TextRange};
+
+declare_rule! {
+    /// Enforce proper usage of `new` and `constructor`.
+    ///
+    /// In JavaScript, classes utilize the `constructor` method to initialize a new instance. On the other hand, TypeScript interfaces can describe a class type with a `new()` method signature, though this pattern is not commonly seen in real-world code. Developers, especially those new to JavaScript or TypeScript, might occasionally confuse the use of `constructor` with `new`.
+    /// This rule triggers warnings in the following scenarios:
+    /// - When a class has a method named `new`.
+    /// - When an interface defines a method named `constructor` or `new` that returns the interface type.
+    /// - When a type alias has a `constructor` method.
+    ///
+    /// You should not use this rule if you intentionally want a class with a `new` method, and you're confident nobody working in your code will mistake it with an `constructor`.
+    ///
+    /// Source: https://typescript-eslint.io/rules/no-misused-new/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// declare class C {
+    ///   new(): C;
+    /// }
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// interface I {
+    ///   new (): I;
+    ///   constructor(): void;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```ts
+    /// declare class C {
+    ///   constructor();
+    /// }
+    ///
+    /// interface I {
+    ///   new (): C;
+    /// }
+    /// ```
+    ///
+    pub(crate) NoMisusedNew {
+        version: "next",
+        name: "noMisusedNew",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) DeclarationQuery = TsInterfaceDeclaration |  TsTypeAliasDeclaration | JsClassDeclaration | TsDeclareStatement
+}
+
+pub(crate) enum RuleState {
+    ClassMisusedNew(TextRange),
+    InterfaceMisusedNew(TextRange),
+    InterfaceMisusedConstructor(TextRange),
+    TypeAliasMisusedConstructor(TextRange),
+}
+
+impl RuleState {
+    fn message(&self) -> MarkupBuf {
+        match self {
+            RuleState::ClassMisusedNew(_) => (markup! {
+                "Do not use the `new` method in class."
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedNew(_) => (markup! {
+                "Do not use the `new` method in interface."
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedConstructor(_) => (markup! {
+                "Do not use the `constructor` method in interface."
+            })
+            .to_owned(),
+            RuleState::TypeAliasMisusedConstructor(_) => (markup! {
+                "Do not use the `constructor` method in type alias."
+            })
+            .to_owned(),
+        }
+    }
+
+    fn note(&self) -> MarkupBuf {
```

Comment:
This 'notes' seems more helpful than the previous "messages". I think we could use them as main diagnostic? We could add a note in some cases. For example for a misued new in a class, we could ask the user if they mean `constructor`. What do you think?

Reply:
I agree with that. more I'll update it more descriptive.

Reply:
Thanks, I feel that in the existing rules, the title is simplified and explain the detailed reason in the note. 
But useful information should be show fast and we need to define roles for title and note. I think that in the previous rules, title stands for what, note for why.

Reply:
I have no strong opinions, but I think the current implementation is fine